api/build.gradle.kts

```diff
@@ -0,0 +1,169 @@
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.springframework.boot") version "2.1.6.RELEASE"
+    id("io.spring.dependency-management") version "1.0.6.RELEASE"
+    id("org.jetbrains.kotlin.jvm") version "1.3.41"
+    id("org.jetbrains.kotlin.plugin.spring") version "1.3.41"
+    id("org.jetbrains.kotlin.kapt") version "1.3.41"
+    id("com.github.ben-manes.versions") version "0.21.0"
+    id("io.gitlab.arturbosch.detekt") version "1.0.0-RC16"
+    id("nu.studer.jooq") version "3.0.3"
+    id("jacoco")
+    id("java")
+    id("idea")
+}
+
+defaultTasks("bootRun")
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+dependencyManagement {
+    imports {
+        mavenBom("org.springframework.cloud:spring-cloud-dependencies:Greenwich.RELEASE")
+    }
+}
+
+tasks.getByName<KotlinCompile>("compileKotlin") {
+    kotlinOptions.jvmTarget = "1.8"
+    kotlinOptions.freeCompilerArgs += listOf("-Xjsr305=strict")
+}
+
+tasks.getByName<KotlinCompile>("compileTestKotlin") {
+    kotlinOptions.jvmTarget = "1.8"
+    kotlinOptions.freeCompilerArgs += listOf("-Xjsr305=strict")
+}
+
+val integrationTestCompile by configurations.creating {
+    extendsFrom(configurations["testCompile"])
+}
+
+val integrationTestRuntime by configurations.creating {
+    extendsFrom(configurations["testRuntime"])
+}
```

Comment:
Could we use

```kts
val integrationTestImplementation by configurations.creating {
    extendsFrom(configurations["testImplementation"])
}
```

instead of

```kts
val integrationTestCompile by configurations.creating {
    extendsFrom(configurations["testCompile"])
}

 val integrationTestRuntime by configurations.creating {
    extendsFrom(configurations["testRuntime"])
}
```

Note: Not sure if it would work

- [ ] `integrationTestRuntime`?